src/sonarqube-gui-plugin/src/main/resources/static/js/cloneViewer/CloneClassSelector.js

```diff
@@ -0,0 +1,172 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL FrontEndART Software Ltd. BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+SM.CloneClassSelector = function(HTMLelem, options) {
+  SM.Subscribable.call(this); // inherit from SM.Subscribable
+  this.elem = null; // :DOMElement
+  this.cloneClassList; // :CloneClass[]
+  this.selected; // :int
+
+  var self = this;
+
+  this.init = function(HTMLelem,options) {
+    this.cloneClassList = (typeof options.cloneClassList !== "undefined") ? options.cloneClassList : [];
+    this.selected = (typeof options.selected !== "undefined") ? options.selected : undefined;
+
+    this.elem = HTMLelem;
+    this.bindElement(HTMLelem);
+    this.renderAll();
+    SM.MetricLoader.subscribe("finishedAllRequests", function() {
+      self.renderMetrics();
+    });
+    this.renderAll();
+  };
+
+  /**
+   * Generates HTML of dependant of the current state of the object.
+   * Then overwrites the HTML in the visible webpage as well.
+   * @return {void}
+   */
+  this.renderAll = function() {
+    this.elem.html("");
+
+    // put together the selectors html string
+    var html = [];
+    html.push('<div>');
+    html.push('<select id="cloneClassSelector">')
+    this.cloneClassList.forEach(function(cloneClass, i) {
+        html.push("<option value="+i+"> "+cloneClass.name+"</option>");
+    });
+    html.push('</select>');
+    html.push('</div>');
+    html.push('<div id="cloneClassMetricsContainer"></div>');
+
+    this.elem.append(html.join(""));
+
+    this.renderMetrics();
+
+    // make selectmenu
+    $("#cloneClassSelector").selectmenu();
+    $("#cloneClassSelector").val(this.selected).selectmenu("refresh");
+
+  };
+
+  this.renderMetrics = function() {
+    var div = $('#cloneClassMetricsContainer');
+    var html = [];
+    var data = this.cloneClassList[this.selected].cloneClassMetrics;
+    var metrics = SM.state[SM.options.component.key].clone.classMetrics;
+    metrics.forEach(function(metric) {
+      html.push(SM.cloneViewer.getFormatedMetric(data[metric.title], metric));
+    });
+    div.html(html.join(""));
+    $("#cloneClassSelectorContainer .sm-cloneviewer-metric-title-container").tooltip({
+      content: function() {
+        return $(this).prop('title');
+      }
+    });
+  };
+
+  /**
+   * selects a CloneClass, and calls everything that needs to update its state.
+   * @param  {int} choice    the id of the cloneclass in the this.CloneClassList
+   * @return {void}
```

Comment:
Please add a new line before `@return` and write `undefined` instead of void like before.